File: code/innobyte/group-53/Stack.php
/**
 * Stack data structure task implementation.
 * Glory Ogar
 * 20/CSC/213
 */
class StackDS
{
    private $stack; // Private property to hold the stack elements
    private $top;   // Private property to keep track of the top of the stack

    /**
     * Constructor to initialize the stack.
     */
    public function __construct()
    {
        $this->stack = []; // Initialize the stack as an empty array
        $this->top = -1;   // Initialize the top pointer to -1 (empty stack)
    }

    /**
     * Push an element onto the stack.
     * @param mixed $value The value to push onto the stack.
     */
    public function push($value)
    {
        $this->top++;             // Increment the top pointer
        $this->stack[$this->top] = $value; // Add the value to the top of the stack
    }

    /**
     * Pop the top element from the stack and return it.
     * @return mixed|null The popped element, or null if the stack is empty.
     */
    public function pop()
    {
        if ($this->isEmpty()) {
            return null; // Return null if the stack is empty
        } else {
            $value = $this->stack[$this->top]; // Get the top element
            unset($this->stack[$this->top]);   // Remove the top element
            $this->top--;                      // Decrement the top pointer
            return $value;                     // Return the popped element
        }
    }

    /**
     * Peek at the top element of the stack without removing it.
     * @return mixed|null The top element, or null if the stack is empty.
     */
    public function peek()
    {
        if ($this->isEmpty()) {
            return null; // Return null if the stack is empty
        } else {
            return $this->stack[$this->top]; // Return the top element
        }
    }

    /**
     * Check if the stack is empty.
     * @return bool True if the stack is empty, false otherwise.
     */
    public function isEmpty()
    {
        return $this->top === -1; // Check if the top pointer is -1
    }
}

File: code/innobyte/group-52/Stack.py
# Stack Data Structure task implementation in Python Programming language
# Author : Emediong Sunday
# Reg No: 20/CSC/033

#Stack class
class Stack:
    def __init__(self):
        """Initialize an empty stack."""
        self.items = []
        
def is_empty(self):
        """Check if the stack is empty."""
        return len(self.items) == 0

    def push(self, item):
        """Push an item onto the top of the stack."""
        self.items.append(item)

    def pop(self):
        """Remove and return the item at the top of the stack."""
        if not self.is_empty():
            return self.items.pop()
        else:
            raise IndexError("pop from an empty stack")
